attest/verifier/types/src/convert/enclave_report_data_contents.rs

```diff
@@ -67,12 +67,9 @@ impl Digestible for prost::EnclaveReportDataContents {
             key,
             custom_identity,
         } = self;
-        transcript.append_primitive(context, b"nonce", nonce);
-        transcript.append_primitive(context, b"key", key);
-        // Since custom identity is optional we only include it if it has data.
-        if !self.custom_identity.is_empty() {
-            transcript.append_primitive(context, b"custom_identity", custom_identity);
-        }
+        nonce.append_to_transcript(context, transcript);
+        key.append_to_transcript(context, transcript);
+        custom_identity.append_to_transcript_allow_omit(context, transcript);
```

Comment:
I had previously mentioned the "hesitation" about using `Vec<u8>`'s Digestible implementation, but I think it's a false assurance to manually call `append_primitive`
While we're using `Self{ a, b, c}` nothing guarantees that `a` is really an `A` type it could be any type since the compiler will elide it.  That's why I updated the test below to manually build up a digest and compare those, to help ensure that the order is really correct.
That's also why I'm no longer testing the difference of each field as the test building the digest ensures there are 2 places adding each field.

Reply:
NVM, the context should have the field name, I misunderstood the usage, fixed now.